File: Python/0383_ransom_note/solution.py
def can_construct(ransomNote,magazine):
    ransomNote = list(ransomNote)
    for letter in magazine:
        if letter in ransomNote:
            ransomNote.remove(letter)
            if len(ransomNote) == 0:
                return True
    return False
    
ransomNote = "bg"
magazine = "efjbdfbdgfjhhaiigfhbaejahgfbbgbjagbddfgdiaigdadhcfcj"
print(can_construct(ransomNote,magazine))

File: JavaScript/0383_ransom_note/solution.js
function can_construct (ransomNote,magazine) {
    ransomNote = Array.from(ransomNote);
    for (let letter of magazine) {
        if (ransomNote.includes(letter)) {
            const index = ransomNote.indexOf(letter);
            ransomNote.splice(index,1);
        }
    }
    return ransomNote.length === 0;
}

const ransomNote = "bg";
const magazine = "efjbdfbdgfjhhaiigfhbaejahgfbbgbjagbddfgdiaigdadhcfcj";
console.log(can_construct(ransomNote,magazine))